reports/agent-that-syncs-pods-and-google-sheets.md

```diff
@@ -0,0 +1,178 @@
+names of tools/libraries created, repos, and so on.
+-->
+
+We created an [agent](https://github.com/SolidLabResearch/google-sheet-sync/releases/tag/v1.0.0) that 
+syncs data between a Google Sheet and a resource on a Solid pod.
+It has the following features:
+
+- Users define which Google Sheet and resource the agent syncs.
+- Users define via a SPARQL query what data the agent selects from the resource.
+- Users define via YARRRML rules how the data from the sheet is converted to RDF before it is stored in the resource.
+- The agent overwrites the sheet with all the queried data when it launches.
+- The agent periodically checks the sheet for changes and stores these changes in the resource.
+- The agent only works with public resources.
+
+You find a screencast of the agent [here](https://cloud.ilabt.imec.be/index.php/s/eFrEKF2YCkSx22j).
+
+<!--
+Provide a list of important technical decisions and assumptions.
+-->
+We made the following important technological decisions and assumptions:
+- We did not include Solid authentication in this agent 
+  because we have done authentication in other challenges, and 
+  it is not crucial for this challenge.
+  [This](https://github.com/SolidLabResearch/google-sheet-sync/issues/13) is the corresponding issue to implement this.
+
+## User flow
+
+<!--
+Describe a concrete user flow with the approved solution.
+Complete the following sections:
+-->
+
+### Actors/actresses
+
+- User of the application.
+
+### Preconditions
+
+- The user has Node.js installed.
+- The user has set a Client ID and Secret for the Google API.
+  See [these instructions](https://github.com/SolidLabResearch/google-sheet-sync#google-sheet-api).
+- A Google Sheet that the agent can sync to.
+
+### Steps
+
+1. Clone the [repository](https://github.com/SolidLabResearch/google-sheet-sync/) (v1.0.0) via 
+   ```shell
+   git clone -b v1.0.0 https://github.com/SolidLabResearch/google-sheet-sync.git
+   ```
+2. Install the dependencies via
+   ```shell
+   npm i
+   ```
+3. Start the authentication server that enables us to give access to a Google account via
+   ```shell
+   npm run auth
+   ```
+   The server is ready when the following message appears in the terminal:
+   ```
+   Authentication server running on port 8081
+   ```
+   We will use this account to access the aforementioned Google Sheet.
+4. Navigate to <http://localhost:8081> in the browser.
+5. Click on the "Authenticate" button.
+6. Log in with the Google account that has access to the aforementioned Google Sheet.
+7. Click on "Continue" when you get the message "Google hasn't verified this app".
+8. Click on "Continue" when you get the message "Sheet Solid sync wants access to your Google Account".
+9. If the authentication succeeded, then you get the message "Authentication successful. Credentials written to credentials.json."
+10. Stop the sever via CTRL + C.
+11. Prepare the local instance of the Community Solid Server (CSS) via
```

Comment:
This is only relevant for demonstrator purposes, right? Is it possible to run this agent with an existing pod?